src/rpc/common/Concepts.h

```diff
@@ -48,10 +49,18 @@ concept Requirement = requires(T a) {
  * as per boost::json documentation for these functions.
  */
 // clang-format off
+template <typename T>
+concept CoroutineProcess = requires(T a, typename T::Input in, typename T::Output out, boost::asio::yield_context* y) {
+    { a.process(in,y) } -> std::same_as<HandlerReturnType<decltype(out)>>;};
+
+template <typename T>
+concept NonCoroutineProcess = requires(T a, typename T::Input in, typename T::Output out) {
+    { a.process(in) } -> std::same_as<HandlerReturnType<decltype(out)>>;};
+
 template <typename T>
 concept HandlerWithInput = requires(T a, typename T::Input in, typename T::Output out) {
-    { a.spec() } -> std::same_as<RpcSpecConstRef>;
-    { a.process(in) } -> std::same_as<HandlerReturnType<decltype(out)>>;}
+    { a.spec() } -> std::same_as<RpcSpecConstRef>;}
+    &&(CoroutineProcess<T>||NonCoroutineProcess<T>)
```

Comment:
Please add some spaces to keep the formatting consistent. I'd also add a space on line 62 between `;}` as well as do that on other lines above.  Also you can use `or` instead of `||` (also `and` instead of `&&`) when it makes code more readable - feel free to use it whenever you think it makes code read like english instead of machine code.

```cpp
and (IsCoroutineProcess<T> or IsNonCoroutineProcess<T>)
```

This reads much nicer IMHO. Hope you can agree 👍